### PR TITLE
Schema label checksum

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The schema migration calls for storing a schema ID+checksum.  The schema ID enum is already defined but we still need serialization and parsing code.

# Changes
- Add a type corresponding to serializing the `SchemaLabel`.  The serialized data includes a checksum.
- Add serializing and parsing functions.

# Tests
- Unit tests are included.

# Todos

- [ ] Add entry to changelog (if necessary).
  None, as the new types are unused.

### CODE REVIEW

**Review by lmuntaner — COMMENTED** [review]
Is the next step to add support for more `SchemaLabel`s that use the new functionality?

**Comment by bitdivine** [comment]
@lmuntaner Regarding more schema labels, there is a parallel PR that introduces the paginated schema, and we will soon have a non-paginated schema as well as the new stable structures are out.

**Comment by lmuntaner** [comment]
@bitdivine I'll leave the technical feedback on the Rust code to @frederikrothenberger and @przydatek

File: rs/backend/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,8 @@ serde = "1.0.136"
 serde_bytes = "0.11.5"
 serde_cbor = "0.11.2"
 sha2 = "0.10.6"
+strum = "0.24.1"
+strum_macros = "0.24.3"
```
[the remark this answers is not in the document]
Comment (frederikrothenberger) [reply rs/backend/Cargo.toml:19]: TIL, that's cool. 👍

**Comment by bitdivine** [comment]
@frederikrothenberger Thank you for the review.

Re why: First some context: We want to store the schema version in raw memory bytes 0-60 (or whatever it is) and use managed memory for later memory.  This way the versioning has pretty much no dependencies.  Candid changes?  We can cope.  Stable memory changes?  Ditto.

But then, what happens if we read bytes 0-60 right now?  We will get the start of the current candid state serialization.  We don't want to accidentally interpret that as a schema version number.  We would have to be unlucky for it to be a legal version number but with a checksum, especially one with a domain separator, we have to be really really unlucky for anything to go wrong.  We can be pretty confident that the bytes we are looking at really are a schema version.

**Pull request merged**